src/VisualStudio/Core/Def/Implementation/Options/RoamingVisualStudioProfileOptionPersister.cs

```diff
@@ -134,6 +133,19 @@ public bool TryFetch(OptionKey optionKey, out object value)
                     return false;
                 }
             }
+            else if (optionKey.Option.Type ==typeof(NamingStylePreferences))
+            {
+                // We store these as strings, so deserialize
+                if (value is string serializedValue)
+                {
+                    value = NamingStylePreferences.FromXElement(XElement.Parse(serializedValue));
```

Comment:
This should be resilient to if that parse fails. It's the responsibility of the persister to deal with it.